src/colab/Initialize_SKAI_Colab_Kernel.ipynb

```diff
@@ -0,0 +1,249 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "Initialize SKAI Colab Kernel",
+      "provenance": [],
+      "private_outputs": true,
+      "collapsed_sections": [
+        "KzSkXvGX6IAz"
+      ],
+      "history_visible": true,
+      "include_colab_link": true
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "language_info": {
+      "name": "python"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "view-in-github",
+        "colab_type": "text"
+      },
+      "source": [
+        "<a href=\"https://colab.research.google.com/github/google-research/skai/blob/skai-colab-0000002/src/colab/Initialize_SKAI_Colab_Kernel.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Initialize SKAI Colab Kernel\n",
+        "This notebook contains a set of commands to run on a new custom colab kernel VM. These commands will pre-install all the necessary dependencies in order to run the SKAI pilot notebook.\n",
+        "\n",
+        "To use:\n",
+        "1. **Open notebook in Playground mode. This is so that you don't accidentally save settings changes you make.**\n",
+        "2. Connect to the custom kernel you want to initialize.\n",
+        "3. Change settings as appropriate.\n",
+        "4. Run all code cells.\n"
+      ],
+      "metadata": {
+        "id": "ahFVM6s5B14O"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#@title Settings\n",
+        "\n",
+        "#@markdown Please enter the parameters of **google cloud platform account** to use:\n",
+        "CLOUD_PROJECT_NAME = \"\" #@param {type:\"string\"}\n",
+        "CLOUD_LOCATION = \"\" #@param {type:\"string\"}\n",
+        "CLOUD_SERVICE_ACCOUNT = \"\" #@param {type:\"string\"}\n",
+        "\n",
+        "#@markdown ---\n",
+        "#@markdown Following parameters are used to clone the **git repository** and set up the **colab workspace directory**.\n",
+        "\n",
+        "#@markdown **We recommend to not change those parameters.**\n",
+        "PRIVATE_KEY_PATH = \"/root/service-account-private-key.json\" #@param {type:\"string\"}\n",
+        "VIRTUALENV_DIR = \"/content/skai_env\" #@param {type:\"string\"}\n",
+        "SKAI_CODE_DIR = \"/content/skai_src\"  #@param {type:\"string\"}\n",
+        "SKAI_REPO = \"https://github.com/google-research/skai.git\"  #@param {type:\"string\"}\n",
+        "SKAI_BRANCH = \"main\" #@param {type:\"string\"}\n",
+        "SKAI_COMMIT = \"14e6da56489da5fd803d3cc7c39b63edef71591e\" #@param {type:\"string\"}"
```

Comment:
This should be empty by default.

Reply:
I had this debat. Objectively, there no reason for the user to change these settings. Also to avoid any difference between those set here and those one used in the other colab that run  the pipeline. Also I specified the commit to avoid any bug in the future when new features/changes will be pushed into the main sec and diverge with the commands implemented in the colab. Will be happy to hear about the best way you think we can fill out those parameters.

Reply:
```suggestion
        "SKAI_COMMIT = \"\" #@param {type:\"string\"}"
```

Actually I am only suggesting that "SKAI_COMMIT" be empty by default. I understand the reasoning for pegging to a specific commit to avoid bugs, but at the same time we would have to remember to update the commit number whenever we commit anything that we want the colab users to have, which IMHO is more of a maintenance burden.

I would suggest that we keep it empty for now, since it's very low harm. If we do experience a lot of breakage down the line, we can add an automated test that tells us when a commit breaks the notebook.

Reply:
I got your point. Let's keep it empty. I will commit a change that handles this blank condition. Thank you for sharing your suggestion.